manifest.yml.tmpl

```diff
@@ -0,0 +1,38 @@
+---
+applications:
+  - name: stub-idp-$ENV
+    routes:
+      - route: stub-idp-$ENV.cloudapps.digital
+    stack: cflinuxfs3
+    memory: 1G
+    buildpack: java_buildpack
+    env:
+      JBP_CONFIG_OPEN_JDK_JRE: '{ jre: { version: 11.+ } }'
+      LOG_PATH: /app/stub-idp/logs
+      LOG_LEVEL: INFO
+      CONFIG_FILE: /app/stub-idp/stub-idp.yml
+      ENTITY_ID: "http://stub_idp.acme.org/%s/SSO/POST"
+      KEY_TYPE: encoded
+      STUB_IDP_SIGNING_PRIVATE_KEY: $SIGNING_KEY
+      CERT_TYPE: encoded
+      STUB_IDP_SIGNING_CERT: $SIGNING_CERT
+      STUB_IDP_BASIC_AUTH: true
+      STUB_IDPS_FILE_PATH: /app/stub-idp/resources/$ENV/stub-idps.yml
+      GRAPHITE_REPORTING_FREQUENCY: 100days
+      METADATA_ENTITY_ID: $METADATA_ENTITY_ID
+      METADATA_URL: $METADATA_URL
+      TRUSTSTORE_TYPE: encoded
+      METADATA_TRUSTSTORE: $METADATA_TRUSTSTORE
+      TRUSTSTORE_PASSWORD: $TRUSTSTORE_PASSWORD
+      EUROPEAN_IDENTITY_ENABLED: $EUROPEAN_IDENTITY_ENABLED
+      HUB_CONNECTOR_ENTITY_ID: $HUB_CONNECTOR_ENTITY_ID
+      STUB_IDP_HOSTNAME: $STUB_IDP_HOSTNAME
+      STUB_COUNTRY_SIGNING_PRIVATE_KEY: $STUB_COUNTRY_SIGNING_PRIVATE_KEY
+      STUB_COUNTRY_SIGNING_CERT: $STUB_COUNTRY_SIGNING_CERT
+      DB_URI: $DB_URI
```

Comment:
shouldn't the database URI come from a service binding rather than from a manifest environment variable?

Reply:
Do you know a way to get a database URI from a service binding? I tried to get it and it failed.

Reply:
You would probably need to change the Stub IDP application to pick out the substring you need from the service binding then substitute it into the application's configuration using a variation of `EnvironmentVariableSubstitutor`. Alternatively, you could add a new of configuring the DB directly from the service binding string. I think the second option is the better one.

In the past we've worked around it using deployment scripts like [here](https://github.com/alphagov/verify-local-matching-service-example/blob/master/deployment.sh#L37) but these don't really fit nicely with the concourse paas deployment model with cf-resource.

Reply:
Thanks. The Stub IDP application already has an ability to get DB configuration details from the service binding. I have added it to a separate commit.